h2o-core/src/main/java/water/fvec/NewChunk.java

```diff
@@ -1170,32 +1186,42 @@ else if(!isNA2(i)){
       assert l!=0 || x==0:"l == 0 while x = " + x + " ms = " + _ms.toString();      // Exponent of zero is always zero
       long t;                   // Remove extra scaling
       while( l!=0 && (t=l/10)*10==l ) { l=t; x++; }
+      BigInteger ll = BigInteger.valueOf(l).multiply(pow10bi(x));
```

Comment:
Can we figure out a faster way to check if the number will fit in long? Ideally something that doesn't involve BigInteger operation.